uplink/src/base/actions.rs

```diff
@@ -19,6 +24,9 @@ pub struct Action {
     pub name: String,
     // action payload. json. can be args/payload. depends on the invoked command
     pub payload: String,
+    // Instant at which action must be timedout
+    #[serde(skip, default = "now")]
+    pub deadline: Instant,
```

Comment:
If this is just for initialization, make this an `Option` and default to `None`

Reply:
Or if Option is unnecessary because of compulsory timeout (due to our design), expose default timeout in config and use it instead of `Instant::now`.

Reply:
All actions have a deadline, except for tunshell, which is the only exception, and hence why we don't do options here.

Reply:
Have reverted to Optional deadline